packages/strapi-plugin-upload/admin/src/components/CardPreview/index.js

```diff
@@ -1,22 +1,37 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+import { getType } from '../../utils';
+
+import BrokenFile from '../../icons/BrokenFile';
 import FileIcon from '../FileIcon';
 import Wrapper from './Wrapper';
 import Image from './Image';
 
-const CardPreview = ({ url, type }) => {
+const CardPreview = ({ hasError, url, type }) => {
+  const isFile = getType(type) === 'file';
+
+  const renderFile = () => <FileIcon fileType={type} />;
+
+  const renderImage = () => <Image src={url} />;
+
```

Comment:
To avoid multiple conditions here, maybe you can add an if statement.
```js
if (hasError) {
    return (
         <Wrapper isFile={...}>
              <BrokenFile />
         </Wrapper>
     );
}

return (
     <Wrapper isFile={isFile}>
            {isFile ? <FileIcon fileType={type} /> : <Image src={url} />}
    </Wrapper>
);
```